spec/lib/orbf/rules_engine/data/contract_spec.rb

```diff
@@ -133,6 +133,117 @@ def test_overlaps(other_start, other_end, expected)
     end
   end
 
+  describe "#initialize" do
+    describe "#validate!" do
+      describe "missing orgunit" do
+        it "raises ContractValidationException with custom message on missing orgunit" do
+          field_values = {
+            "id"                  => "GPAGzdsLDTP",
+            "date"                => "2018-06-01T00:00:00.000+0000",
+            "contract_end_date"   => "2021-12-31",
+            "contract_type"       => "PMA"
+          }
+      
+          missing_attrs = "contract_start_date, org_unit"
+          
+          expect { Orbf::RulesEngine::Contract.new(field_values, calendar) }.to raise_error(Orbf::RulesEngine::ContractValidationException)
```

Comment:
Again, personal thing here, but I'll explain my reasoning!

I don't like multiple `expect` in a single test because they tend to be more brittle. If we change the Class of the error we now have to update it in multiple tests. So I would write one test to check that it raises the Error we expect, and then in the other tests it would rely on raise_error.with_message (since I don't care about the error class in those tests, just that it matches the message)

Since we've moved some logic into the error class (always need to be careful when doing that, because if we make an error inside the Error, we lose our nice error), I would test the Error instead of the `Contract` class.

Like

``` ruby
describe ContractValidationException do
   it "has custom messages on missing org unit attributes" do
      exception = ContractValidationException .new(["org_unit"], {"name" => "", "id" => ""})
      
      except(exception.message).to match(/#{relevant_org_unit_id}/)
  end
end
```

And the in the contract spec, I'd just check that it raises the error like we expect. That way if we modify the behavior of the Error we only need to update the specs for that error.

Again, all this is more a personal style thing (but imho makes for less brittle tests), so I'm good with the code as is, I think we're doing exactly what we wanted it to do.